modules/core/src/main/scala/com.snowplowanalytics/iglu.schemaddl/jsonschema/SelfSyntaxChecker.scala

```diff
@@ -96,6 +96,43 @@ object SelfSyntaxChecker {
       |"additionalProperties": false,
       |"default":{}}""".stripMargin
 
+  private val SelfSchemaText = {
+    val vendorRegex = "^[a-zA-Z0-9-_.]+$"
```

Comment:
`.` would match everything. It needs escaping.

Reply:
`.` inside a `[]` does not match everything; it only matches an explicit dot.  I only learnt this recently!

But I also agree that it's confusing and non-obvious, so I will add escaping for better clarity.

Reply:
This is something I did not know. Checking, what other things `[]` escape.